server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -53,6 +53,33 @@ protected static Set<String> getEnabledRemoteClusters(final Settings settings) {
         return RemoteConnectionStrategy.getRemoteClusters(settings);
     }
 
+    public static boolean isRemoteIndexName(String indexName) {
+        if (indexName.isEmpty() || indexName.charAt(0) == '<' || indexName.startsWith("-<")) {
+            // This is date math, but eve if it is not, the remote can't start with '<'.
+            // Thus, whatever it is, this is definitely not a remote index.
+            return false;
+        }
+        // Note remote index name also can not start with ':'
+        return indexName.indexOf(RemoteClusterService.REMOTE_CLUSTER_INDEX_SEPARATOR) > 0;
+    }
+
+    public static String[] splitIndexName(String indexName) {
+        if (indexName.isEmpty() || indexName.charAt(0) == '<' || indexName.startsWith("-<")) {
+            // This is date math, but eve if it is not, the remote can't start with '<'.
+            // Thus, whatever it is, this is definitely not a remote index.
+            return new String[] { null, indexName };
+        }
+        int i = indexName.indexOf(RemoteClusterService.REMOTE_CLUSTER_INDEX_SEPARATOR);
+        if (i == 0) {
+            throw new IllegalArgumentException("index name [" + indexName + "] is invalid because the remote part is empty");
+        }
+        if (i < 0) {
+            return new String[] { null, indexName };
+        } else {
+            return new String[] { indexName.substring(0, i), indexName.substring(i + 1) };
```

Comment:
Hard to say because:
1. We don't know what this `:` belongs too (may be some other syntax)
2. Oddly enough, some code assumes empty index is ok, so I'm not feeling comfortable to put an error here, even though I suspect it's not actually a valid index name.